test/apps/iodemo/io_demo.cc

```diff
@@ -1772,6 +1790,12 @@ class DemoClient : public P2pDemoCommon {
 
         assert(conn->ucx_status() == UCS_OK);
 
+        if (opts().validate) {
+            assert(length == opts().iomsg_size);
+            check_conn_id(msg, conn);
+            validate(conn, msg, opts().iomsg_size);
```

Comment:
logically we have to check `length` in spite of it's equal to `opts().iomsg_size`

Reply:
done